2024/12/solve-1.ys

```diff
@@ -1,100 +1,72 @@
 #!/usr/bin/env ys-0
 
-defn main(& args):
-  terrain =: init(args.0 || "input")
-  regions =: terrain:regions
-  println:
-    total-price: terrain regions
-
-defn init(file="input"):
-  file:slurp:lines
+defn main(input='input'): !:say
+  terrain =: input:slurp:lines
+  regions =: regions(terrain)
+  total-price: terrain regions
 
 defn total-price(terrain regions):
-  sum:
-    map _ regions:
-      fn(r): price(terrain r)
-
-defn price(terrain region):
-  =>: region:len * perimeter(terrain region)
+  defn price(region):
+    region.# * perimeter(terrain region)
```

Comment:
`foo.#` == `foo:len` == `(count foo)`

Reply:
`=>:` is useful when you need a pair and only have a scalar, but we didn't need a pair here.
If we did need a pair, you could also do:
```
region.# *: perimeter(terrain region)
```